imap_processing/idex/idex_packet_parser.py

```diff
@@ -89,6 +91,45 @@ def __init__(self, packet_file: str):
 
         self.data = xr.concat(processed_dust_impact_list, dim="Epoch")
 
+    def write_l1_cdf(self, description: str = "", version: str = "01", directory=""):
+        """Write the contents of self.data to a CDF file.
+
+        The date in the file name is determined by the start time of the first event
+        in the file
+
+        Parameters
+        ----------
+            description (str): The description to insert into the file name after the
+                               orbit, before the SPICE field.  No underscores allowed.
+            version (str):  The version number to append to the file
+            directory (str): The directory to write the file to
+
+        Returns
+        -------
+            str
+                The name of the file created
+        """
+        # Determine the start date of the data in the file,
+        # based on the time of the first dust impact
+        file_start_date = self.data["Epoch"][0].data
+        date_string = np.datetime_as_string(file_start_date, unit="D").replace("-", "")
+        if description:  # Just append an underscore to the description
+            description = f"_{description}"
```

Comment:
To avoid double underscore
```suggestion
            description = description if description.startswith("_") else f"_{description}"
```